pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -1266,20 +1266,45 @@ func Convert_v1_Firmware_To_related_apis(vmi *v1.VirtualMachineInstance, domain
 	}
 
 	if firmware.ACPI != nil {
-		if slicNameRef := firmware.ACPI.SlicNameRef; slicNameRef != "" {
-			path := fmt.Sprintf("/var/run/kubevirt-private/secret/%s/slic.bin", slicNameRef)
-			domain.Spec.OS.ACPI = &api.OSACPI{
-				Table: api.ACPITable{
-					Type: "slic",
-					Path: path,
-				},
-			}
+		path, err := getSlicMountedPath(vmi.Spec.Volumes, firmware.ACPI.SlicNameRef)
+		if err != nil {
+			log.Log.Object(vmi).Warningf("Failed to get supported path for Volume: %s", firmware.ACPI.SlicNameRef)
+			return err
+		}
+
+		domain.Spec.OS.ACPI = &api.OSACPI{
+			Table: api.ACPITable{
+				Type: "slic",
+				Path: path,
+			},
 		}
 	}
 
 	return nil
 }
 
+func getSlicMountedPath(volumes []v1.Volume, name string) (string, error) {
+	sourcePath := ""
+	// We need to know the the volume type referred by @name
+	for _, volume := range volumes {
+		if volume.Name != name {
+			continue
+		}
+
+		if volume.Secret != nil {
+			sourcePath = config.GetSecretSourcePath(name)
+			break
+		}
+	}
```

Comment:
When we find the volume we should break unconditionally

Reply:
It was made like that considering future changes, if we would support other volume types.
I've moved the logic inside the loop, to return success/failure when volume is found and an extra likely nop error in case volume isn't found. Diff:

```diff
--- a/pkg/virt-launcher/virtwrap/converter/converter.go
+++ b/pkg/virt-launcher/virtwrap/converter/converter.go
@@ -1253,25 +1253,22 @@ func Convert_v1_Firmware_To_related_apis(vmi *v1.VirtualMachineInstance, domain
 }

 func getSlicMountedPath(volumes []v1.Volume, name string) (string, error) {
-       sourcePath := ""
        // We need to know the the volume type referred by @name
        for _, volume := range volumes {
                if volume.Name != name {
                        continue
                }

-               if volume.Secret != nil {
-                       sourcePath = config.GetSecretSourcePath(name)
-                       break
+               if volume.Secret == nil {
+                       return "", fmt.Errorf("Firmware's slic volume type is unsupported")
                }
-       }

-       if sourcePath == "" {
-               return "", fmt.Errorf("Firmware's slic volume type is unsupported")
+               // Return path to slic binary data
+               sourcePath := config.GetSecretSourcePath(name)
+               return filepath.Join(sourcePath, "slic.bin"), nil
        }

-       // Return path to slic binary data
-       return filepath.Join(sourcePath, "slic.bin"), nil
+       return "", fmt.Errorf("Firmware's slic volume type not found")
 }
```